components/relevancy/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Proposed API for the relevancy component (validation phase)
+//!
+//! The goal here is to allow us to validate that we can reliably detect user interests from
+//! history data, without spending too much time building the API out.  There's some hand-waving
+//! towards how we would use this data to rank search results, but we don't need to come to a final
+//! decision on that yet.
+
+/// List of possible interests for a domain.  Each domain would have exactly one of these.  If we
+/// can't calculate an interest based on the classifier data, then we put it in `Unknown`
+#[derive(uniffi::Enum)]
+pub enum Interest {
+    Education,
+    Gaming,
+    Sports,
+    Unknown,
+}
+
+/// An interest combined with a relevancy score
+///
+/// Scores will be [0-10].  Consumers would use this score to alter the order of their suggestions.
+///
+/// For example, consumers may assign a default score of 5 to sponsored suggestions and 0 to
+/// Wikipedia suggestions.  By default, the sponsored suggestions will show up over Wikipedia
+/// suggestions, but if a Wikipedia suggestion had a interest that was highly relevant for the
+/// user, that suggestion would show up first instead.
+#[derive(uniffi::Record)]
```

Comment:
I was thinking about trying to test out the new UniFFI proc-macros, but on second thought that's going to cause a lot of issues on Desktop.  I don't actually know how we're going to be able to get those to work with the FF build system and the current JS bindings implementation.

In general I'd like to try out some newer UniFFI features with this component, since there's not that much risk.  If we run into some new UniFFI bug and that causes us to lose some experiment data, that should be fine.  

However, we should prioritize getting the experiment out over quickly over testing out UniFFI features.  So if some UniFFI feature is going to require work on the `gecko-js` bindings, let's not delay the experiments.